src/NUnitFramework/framework/Internal/TestExecutionContext.cs

```diff
@@ -360,6 +365,19 @@ internal IWorkItemDispatcher Dispatcher
         public ParallelScope ParallelScope { get; set; }
 
         /// <summary>
+        /// The unique name of the worker that spawned the context.
+        /// For builds with out the parallel feature, it is null.
+        /// </summary>
+        public string WorkerId {
+            get {
+                if (_workerId != null) return _workerId;
+                if (_priorContext != null) return _priorContext.WorkerId;
+                return null;
+            }
+            internal set {_workerId = value;}
+        }
```

Comment:
I've got to admit, this took a lot more thought then expected to get it passed through at the correct time, but it appears to be working properly.